bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/AccountHandler.java

```diff
@@ -800,4 +862,187 @@ private void handlePushActivity(@Nullable String payload) {
+            discoverSmartHome = ((BigDecimal) discoverSmartHomeConfig).intValue();
+        }
+        return discoverSmartHome;
+    }
+
+    public boolean shouldDiscoverOpenHABSmartHomeDevices() {
+        Configuration config = getThing().getConfiguration();
+        Boolean discoverOpenHabSmartHomeDevices = (Boolean) config.getProperties()
+                .get("discoverOpenHabSmartHomeDevices");
+        if (discoverOpenHabSmartHomeDevices == null) {
+            return false;
+        }
+        return discoverOpenHabSmartHomeDevices;
+    }
+
+    public List<SmartHomeBaseDevice> updateSmartHomeDeviceList(boolean forceUpdate) {
+        Connection currentConnection = connection;
+        if (currentConnection == null) {
+            return Collections.emptyList();
+        }
+
+        if (!forceUpdate && this.smartHomeDeviceHandlers.isEmpty() && shouldDiscoverSmartHomeDevices() == 0) {
+            return Collections.emptyList();
+        }
+
+        List<SmartHomeBaseDevice> smartHomeDevices = null;
+        try {
+            if (currentConnection.getIsLoggedIn()) {
+                smartHomeDevices = currentConnection.getSmarthomeDeviceList();
+            }
+        } catch (IOException | URISyntaxException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getLocalizedMessage());
+        }
+        if (smartHomeDevices != null) {
+            // create new id map
+            Map<String, SmartHomeBaseDevice> newJsonIdSmartHomeDeviceMapping = new HashMap<>();
+            for (Object smartHomeDevice : smartHomeDevices) {
+                if (smartHomeDevice instanceof SmartHomeBaseDevice) {
+                    SmartHomeBaseDevice smartHomeBaseDevice = (SmartHomeBaseDevice) smartHomeDevice;
+                    String id = smartHomeBaseDevice.findId();
+                    if (id != null) {
+                        newJsonIdSmartHomeDeviceMapping.put(id, smartHomeBaseDevice);
+                    }
+                }
+            }
+            jsonIdSmartHomeDeviceMapping.clear();
+            jsonIdSmartHomeDeviceMapping.putAll(newJsonIdSmartHomeDeviceMapping);
+        }
+        // update handlers
+        synchronized (smartHomeDeviceHandlers) {
+            smartHomeDeviceHandlers
+                    .forEach(child -> child.setDeviceAndUpdateThingState(this, findSmartDeviceHomeJson(child)));
+        }
+
+        if (smartHomeDevices != null) {
+            Map<String, SmartHomeDevice> newJsonSerialDeviceMapping = new HashMap<>();
+            for (Object smartDevice : smartHomeDevices) {
+                if (smartDevice instanceof SmartHomeDevice) {
+                    SmartHomeDevice shd = (SmartHomeDevice) smartDevice;
+                    String entityId = shd.entityId;
+                    if (entityId != null) {
+                        newJsonSerialDeviceMapping.put(entityId, shd);
+                    }
+                }
+            }
+        }
```

Comment:
Is this code incomplete? It doesn't look like it does anything.

Reply:
Agreed. I'm not sure if this was intended to enable the AccountServlet to provide more information (like it already does for the echo devices). I have added a field, so the mapping is kept and this can be extended later